### PR TITLE
Improve `__advance_inround` method to do as much work as possible

### LINKED ISSUE #15: Improve `__advance_inround` method to do as much work as possible
While the assumption is currently true, and will likely stay true, writing code that does not rely on the assumption is trivial. See https://github.com/stIncMale/tic-tac-toe/commit/35aa3bf26832835365565b86dc0ef425246b7eb5 for an example on I did this in https://github.com/stIncMale/tic-tac-toe.

### PR DESCRIPTION
**Pull request by Yo1k**

### CODE REVIEW

File: yo1k/tic_tac_toe/test/kernel/test_game.py [hunk 1]
```diff
@@ -190,6 +190,20 @@ def test_advance__ready_action__beginning(self) -> None:
                 step=0)
         self.assertEqual(expected_state, state)
 
+    def test_advance__stop_at__phase_change(self) -> None:
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/test/kernel/test_game.py]: `stop_at__phase_change` -> `stop_at_phase_change`

File: yo1k/tic_tac_toe/test/kernel/test_game.py [hunk 2]
```diff
@@ -190,6 +190,20 @@ def test_advance__ready_action__beginning(self) -> None:
...
+        Logic((
+                ListActionQueue([Action.new_surrender(), Action.new_occupy(Cell(0, 0))]),
+                ListActionQueue([]))) \
+            .advance(state)
+        expected_state = _new_state(
+                player_o=_new_player(mark=Mark.O, wins=1),
+                board=Board([[None, None, None], [None, None, None], [None, None, None]]),
+                phase=Phase.OUTROUND,
+                required_ready=expected_required_ready)
+        self.assertEqual(expected_state, state)
```
Comment (stIncMale) [inline yo1k/tic_tac_toe/test/kernel/test_game.py:207]: Currently, this test relies on the `assert` statement somewhere in the `Logic.advance` method. Let's `pop` from the `ActionQueue` of the `player_x` at the end of the test and confirm that we `pop`ed not `None`.

**Review by stIncMale — CHANGES_REQUESTED** [review]

File: yo1k/tic_tac_toe/test/kernel/test_game.py (hunk 1 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/test/kernel/test_game.py]: Done.

File: yo1k/tic_tac_toe/test/kernel/test_game.py (hunk 2 above)
Comment (Yo1k) [reply yo1k/tic_tac_toe/test/kernel/test_game.py:207]: Done.

**Review by stIncMale — APPROVED** [review]

**Pull request merged**